api/requests/serializers.py

```diff
@@ -209,17 +209,32 @@ def _create_node_request(self, node, validated_data) -> NodeRequest:
         comment = validated_data.get('comment', '')
         requested_permissions = validated_data.get('requested_permissions')
         try:
-            node_request = NodeRequest.objects.create(
-                target=node,
-                creator=creator,
-                comment=comment,
-                machine_state=DefaultStates.INITIAL.value,
-                request_type=request_type,
-                requested_permissions=requested_permissions,
-            )
-            node_request.save()
+            with transaction.atomic():
```

Comment:
Because the error would happen mid-same, you need this transaction to be able to rollback on error.